formstack-consents/src/main/scala/com/gu/identity/formstackconsents/IdentityClient.scala

```diff
@@ -0,0 +1,66 @@
+package com.gu.identity.formstackconsents
+
+import io.circe.syntax._
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent
+import com.gu.identity.formstackconsents.Lambda.{FormstackSubmission, Config}
+import com.typesafe.scalalogging.StrictLogging
+import io.circe.generic.extras._
+import scalaj.http.{Http, HttpResponse}
+import scala.util.Try
+
+class IdentityClient(config: Config) extends StrictLogging {
+
+  val newsletters: List[Newsletter] = List(Holidays, Students, Universities, Teachers, Masterclasses, SocietyWeekly, EdinburghFestivalDataCollection)
+
+  def getResponseFromIdentity(formstackSubmission: FormstackSubmission, newsletter: Newsletter): Option[HttpResponse[String]] = {
+    // The key in the JSON sent to Identity depends on the listType. Sometimes the listType is 'set-lists' with a value of the consent name,
+    // and sometimes the listType is 'set-consents'. See example below.
+    //  {
+    //    "email" : "lauren.emms@guardian.co.uk",
+    //    "set-consents" : "holidays"
+    //  }
+
+    implicit val circeConfig: Configuration = Configuration.default.copy(
+      transformMemberNames = {
+        case "listType" => newsletter.listType
+        case other => other
+      }
+    )
+
+    @ConfiguredJsonCodec case class IdentityRequest(email: String, listType: List[String])
+
+    val requestBody = IdentityRequest(formstackSubmission.emailAddress, List(newsletter.consent))
+
+    Try {
+      Http(s"${config.idapiHost}/consent-email")
+        .headers(("Authorization", config.idapiAccessToken), ("Content-type", "application/json"), ("Accept", "text/plain"))
+        .postData(requestBody.asJson.noSpaces)
+        .asString
+    }.toOption
+  }
+
+  def handleResponseFromIdentity(response: Option[HttpResponse[String]], formstackSubmission: FormstackSubmission, newsletter: Newsletter): Option[APIGatewayProxyResponseEvent] = {
```

Comment:
Don't think the return type should be an `Option` - if the post to the identity API fails, we can encode this information using the `APIGatewayProxyResponseEvent`

Reply:
Or you could keep it as `Option` which would be defined iff the `Option[HttpResponse[String]]` is defined (if the response is non 2xx, then you could use its status to set the status of the `APIGatewayProxyResponseEvent`)

Reply:
Thank you, have kept this as an option and used the request code as you suggested.